spec/format_spec.rb

```diff
@@ -1,55 +1,86 @@
-require_relative '../lib/parser'
+require_relative '../lib/awesome_list'
 require 'net/http'
 require 'faraday'
 
-describe 'formatting' do
+describe 'formatting the list' do
   let(:readme) { File.open("README.md").read }
-  let(:list) { AwesomeListRender.parse(readme) }
+  let(:list) { AwesomeList.parse(readme) }
 
-  describe 'summary display' do
-    context 'sections' do
+  describe 'summary' do
+    context 'sections format' do
       it 'must be in alphabetical order' do
         list.summary.each do |section|
-          titles = section.css('>li>a').map{|a| a.children.text }
+          titles = section.css('>li>a').map { |a| a.children.text.downcase }
           expect(titles).to eq(titles.sort)
         end
       end
     end
 
-    context 'sub sections' do
+    context 'sub sections format' do
       it 'must be in alphabetical order' do
         list.summary.css('>li').each do |subsection|
-          titles = subsection.css('li a').map{|a| a.children.text }
+          titles = subsection.css('li a').map { |a| a.children.text.downcase }
           expect(titles).to eq(titles.sort)
         end
       end
     end
   end
 
-  describe 'projects display' do
+  describe 'projects' do
     context 'sections titles' do
       it 'must be in alphabetical order' do
-        titles = list.section_titles.map { |t| t.text }
+        titles = list.section_titles.map { |t| t.text.downcase }
         expect(titles).to eq(titles.sort)
       end
     end
 
     context 'links' do
       it 'must be in alphabetical order' do
-        list.project_links.each_with_index do |plinks, i|
-          link_texts = plinks.map{|a| a.children.text}
+        list.projects.each_with_index do |project, i|
+          link_texts = project.css('li a').map {|a| a.children.text.downcase }
 
           expect(link_texts).to eq(link_texts.sort),
-            project_sort_error(list.subsection_titles[i], link_texts)
+            project_sort_error(list.subsection_titles[i-1], link_texts)
+        end
+      end
+    end
+
+    context 'descriptions' do
+      it 'must start with captalized letter' do
+        list.projects.css('li').each do |project|
+          description = project.css('p').last.text
```

Comment:
this loop structure is repeated, move it outside to a function that gives an array of descriptions?

Reply:
We can't do it cause `project.css('p')` return 2 items and the last one is the description.

Reply:
Okay, point.

Reply:
👍